shoop/admin/modules/products/views/toolbars.py

```diff
@@ -94,33 +96,66 @@ def _get_variation_and_package_menu_items(self, product):
                     icon="fa fa-eye",
                     url=get_model_url(sib),
                 )
-        elif package_parent:
+
+    def _get_package_menu_items(self, product):
```

Comment:
How about something like this...

``

    def _get_package_url(self, product):
        return reverse("shoop_admin:product.edit_package", kwargs={"pk": product.pk})

    def _get_variation_url(self, product):
        return reverse("shoop_admin:product.edit_variation", kwargs={"pk": product.pk})

    def _get_children_items(children):
        for child in children:
            yield DropdownItem(
                text=_("Child: %s") % child,
                icon="fa fa-eye",
                url=get_model_url(child),
            )

    def _get_parent_and_sibling_items(parent, siblings):
        yield DropdownItem(
            text=_("Parent: %s") % parent,
            icon="fa fa-eye",
            url=get_model_url(parent),
        )
        for sib in siblings:
            yield DropdownItem(
                text=_("Sibling: %s") % sib,
                icon="fa fa-eye",
                url=get_model_url(sib),
            )

    def _get_variation_menu_items(self, product):
        yield DropdownDivider()
        yield DropdownHeader(text=_("Variations"))
        yield DropdownItem(
            text=_("Manage Variations"),
            icon="fa fa-sitemap",
            url=self._get_variation_url(product),
        )
        if product.is_variation_parent():
            self._get_children_items(product.variation_children.all())
        elif product.is_variation_child():
            self._get_parent_and_sibling_items(product.variation_parent, product.get_variation_siblings())

    def _get_package_menu_items(self, product):
        yield DropdownDivider()
        yield DropdownHeader(text=_("Packages"))
        yield DropdownItem(
            text=_("Manage Package"),
            icon="fa fa-cube",
            url=self._get_package_url(product),
        )
        if product.is_package_parent():
            self._get_children_items(product.get_all_package_children())
        elif product.is_package_child():
            for parent in product.get_all_package_parents():
                self._get_parent_and_sibling_items(parent, [sib for sib in parent.get_all_package_children() if sib != product])

    def _get_variation_and_package_menu_items(self, product):
        if (product.is_variation_parent() or product.is_variation_child()):
            for item in self._get_variation_menu_items(product):
                yield item
        elif (product.is_package_parent() or product.is_package_child()):
            for item in self._get_package_menu_items(product):
                yield item
        else:
            yield DropdownDivider()
            yield DropdownHeader(text=_("Packages"))
            yield DropdownItem(
                text=_("Convert to Package Parent"),
                icon="fa fa-retweet",
                url=self._get_package_url(product),
            )
            yield DropdownDivider()
            yield DropdownHeader(text=_("Variations"))
            yield DropdownItem(
                text=_("Convert to Variation Parent"),
                icon="fa fa-retweet",
                url=v_get_variaton_url(product),
            )
``